viewportHax.js

```diff
@@ -3,48 +3,48 @@
 let prevVVheight = window.visualViewport.height;
 
 function onVirtualKeyboard(isUp) {
-    document.body.scrollTop = 0; // For Safari
-    document.documentElement.scrollTop = 0; // For Chrome, Firefox
-    const canvas = document.getElementById("canvas");
-    canvas.style.zIndex = isUp ? -1 : 1;
-    const view = document.getElementById("view");
-    view.style.backgroundColor =  isUp ? "rgba(0, 0, 0, 0.5)" : "transparent";
-    view.style.paddingTop = `${isUp ? 0 : canvas.offsetHeight}px`;
+  document.body.scrollTop = 0; // For Safari
+  document.documentElement.scrollTop = 0; // For Chrome, Firefox
+  const canvas = document.getElementById("canvas");
+  canvas.style.zIndex = isUp ? -1 : 1;
+  const view = document.getElementById("view");
+  view.style.backgroundColor = isUp ? "rgba(0, 0, 0, 0.5)" : "transparent";
+  view.style.paddingTop = `${isUp ? 0 : canvas.offsetHeight}px`;
 
-    const navHeight = document.querySelector("nav").offsetHeight;
-    const footerHeight = document.querySelector("footer").offsetHeight;
-    const vpH = window.visualViewport.height;
-    view.style.height = `${vpH - navHeight - footerHeight}px`;
+  const navHeight = document.querySelector("nav").offsetHeight;
+  const footerHeight = document.querySelector("footer").offsetHeight;
+  const vpH = window.visualViewport.height;
+  view.style.height = `${vpH - navHeight - footerHeight - 15}px`;
```

Comment:
added -15 to add a bit of space below the message box

Reply:
nice i was just thinking the same